server/src/main/java/org/opensearch/common/blobstore/BlobContainer.java

```diff
@@ -124,6 +124,15 @@ default long readBlobPreferredLength() {
      */
     void writeBlob(String blobName, InputStream inputStream, long blobSize, boolean failIfAlreadyExists) throws IOException;
 
+    /**
+     * Needs to be set to true by the implementation if remote data integrity check is supported
+     *
+     * @return true if remote data integrity check is supported
+     */
+    default boolean isRemoteDataIntegritySupported() {
+        return false;
```

Comment:
Do we need this now given we have a `VerifyingMultiStreamBlobContainer`. Once we split the WriteContext we can reintroduce this?

Reply:
We can add an `instanceof` check wherever this call is made and remove this.